pkg/filesystem/local_directory_unix.go

```diff
@@ -13,6 +14,16 @@ import (
 	"google.golang.org/grpc/status"
 )
 
+const (
+	// Permissions do not have one-to-one mapping on Windows. Trivial mapping under Unix.
+	FileMode644 = 0644
```

Comment:
Instead of going into this direction, I'd rather see that `pkg/filesystem` is implemented in such a way that it only provides the actual subset of stuff that we need.

What if we were to change `FileInfo` from this:

```
type FileInfo interface {
	Name() string
	Mode() os.FileMode
}
```

To this:

```
type FileInfo interface {
	Name() string
	Type() os.FileMode // Just the file type; not the permission bits. No need to mask.
	IsExecutable() bool
}
```

Doing this would allow us to implement this in a more platform agnostic way and will also clean up some of the callers.

I suspect that this will also allow us to remove the `Umask()` calls from the unit tests, which were only needed to make exact comparisons against permission bits work.

Reply:
Alternatively: considering that `IsExecutable` only applies to regular file, we could maybe even go so far as to come up with our own enum:

```go
type FileType int

const (
	FileTypeRegularFile FileType = iota
	FileTypeExecutableFile
	FileTypeDirectory
	FileTypeSymlink
	FileTypeUnknown
)

type FileInfo interface {
	Name() string
	Type() FileType
}